src/server/helpers/search.js

```diff
@@ -224,65 +223,71 @@ search.generateIndex = () => {
 	};
 
 	// First, drop index and recreate
-	return _client.indices.delete({index: _index})
-		.catch((err) => {
-			/**
-			 * If the index is missing, don't worry, it probably never existed;
-			 * otherwise, rethrow
- 			 */
-			if (!err.message.startsWith('IndexMissingException')) {
-				throw err;
-			}
-		})
-		// GOTCHA: index creation is buggy
-		// https://tickets.metabrainz.org/browse/BB-205
-		.then(() => _client.indices.create(
-			{index: _index, body: indexMappings}
-		))
-		.then(() => {
-			const baseRelations = [
-				'annotation',
-				'disambiguation',
-				'defaultAlias'
-			];
-
-			const entityBehaviors = [
-				{model: Creator,
-					relations:
-						['gender', 'creatorType', 'beginArea', 'endArea']},
-				{
-					model: Edition,
-					relations: [
-						'publication',
-						'editionFormat',
-						'editionStatus'
-					]
-				},
-				{model: Publication, relations: ['publicationType']},
-				{model: Publisher, relations: ['publisherType', 'area']},
-				{model: Work, relations: ['workType']}
-			];
-
-			// Update the indexed entries for each entity type
-			return Promise.map(entityBehaviors, (behavior) =>
-				behavior.model.forge()
-					.fetchAll({
-						withRelated: baseRelations.concat(behavior.relations)
-					})
-					.then((collection) => collection.toJSON())
-					.map(search.indexEntity)
-			);
-		})
-		.then(() =>
-			Area.forge()
-				// countries only
-				.where({type: 1})
-				.fetchAll()
-				.then((collection) => collection.toJSON())
-				.map(search.indexArea)
-
-		)
-		.then(search.refreshIndex);
+	const mainIndexExists = await _client.indices.exists({index: _index});
+
+	if (mainIndexExists) {
+		await _client.indices.delete({index: _index});
+	}
+
+	await _client.indices.create(
+		{index: _index, body: indexMappings}
+	);
+
+	const baseRelations = [
+		'annotation',
+		'disambiguation',
+		'defaultAlias'
+	];
+
+	const entityBehaviors = [
+		{
+			model: Creator,
+			relations: [
+				'gender',
+				'creatorType',
+				'beginArea',
+				'endArea'
+			]
+		},
+		{
+			model: Edition,
+			relations: [
+				'publication',
+				'editionFormat',
+				'editionStatus'
+			]
+		},
+		{model: Publication, relations: ['publicationType']},
+		{model: Publisher, relations: ['publisherType', 'area']},
+		{model: Work, relations: ['workType']}
+	];
+
+	// Update the indexed entries for each entity type
+	for (const behavior of entityBehaviors) {
+		const collection = await behavior.model.forge()
+			.fetchAll({
+				withRelated: baseRelations.concat(behavior.relations)
+			});
+
+		const entities = collection.toJSON();
+
+		for (const entity of entities) {
+			await search.indexEntity(entity);
+		}
+	}
+
```

Comment:
Same issue that @LordSputnik brought up.

Here's an idea for a solution. It still blocks the subsequent area indexing though.
```
const behaviorPromise = entityBehaviors.map((behavior) =>
    behavior.model.forge().fetchAll({
        withRelated: baseRelations.concat(behavior.relations)
    })
);
const entityList = await Promise.all(behaviorPromise);
const indexedEntities = entityList.map((entities) =>
    Promise.all(entities.toJSON().map((entityObj) => 
        search.indexEntity(entityObj)
    ))
)
await Promise.all(indexedEntities);
    
```